x/campaign/types/params_test.go

```diff
@@ -0,0 +1,84 @@
+package types
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestParamsValidateBasic(t *testing.T) {
+	tests := []struct {
+		name   string
+		params Params
+		err    error
+	}{
+		{
+			name:   "invalid min total supply",
+			params: NewParams(sdk.ZeroInt(), DefaultMaxTotalSupply),
+			err:    errors.New("minimum total supply should be greater than one: invalid total supply range"),
+		},
+		{
+			name:   "min total supply greater than max",
+			params: NewParams(DefaultMaxTotalSupply, DefaultMinTotalSupply),
+			err:    errors.New("maximum total supply should be greater or equal than minimum total supply: invalid total supply range"),
+		},
+		{
+			name:   "valid range",
+			params: NewParams(DefaultMinTotalSupply, DefaultMaxTotalSupply),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := tt.params.ValidateBasic()
+			if tt.err != nil {
+				require.Error(t, err, tt.err)
+				require.Equal(t, err, tt.err)
+				return
+			}
+			require.NoError(t, err)
+		})
+	}
+}
+
+func TestValidateTotalSupplyRange(t *testing.T) {
+	tests := []struct {
+		name        string
+		supplyRange interface{}
+		err         error
+	}{
+		{
+			name:        "invalid interface",
+			supplyRange: "test",
+			err:         fmt.Errorf("invalid parameter type: string"),
+		},
```

Comment:
we should test the `MinTotalSupply` lower than one too

Reply:
we have decided to not fully test validation except where the core functionality is implemented. In all methods using that, we just have a single valid and invalid tests. @lubtd can confirm